x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/bulk_create_suppressed_alerts_in_memory.ts

```diff
@@ -99,6 +136,105 @@ export const bulkCreateSuppressedAlertsInMemory = async ({
+    toReturn,
+    bulkCreate,
+    services,
+    ruleExecutionLogger,
+    tuple,
+    alertSuppression,
+    alertWithSuppression,
+    alertTimestampOverride,
+    experimentalFeatures,
+    maxNumberOfAlertsMultiplier,
+  });
+};
+
+/**
+ * wraps, bulk create and suppress alerts in memory, also takes care of missing fields logic.
+ * If parameter alertSuppression.missingFieldsStrategy configured not to be suppressed,
+ * regular alerts will be created for such events without suppression
+ */
+export const bulkCreateSuppressedSequencesInMemory = async ({
+  sequences,
+  toReturn,
+  wrapSequences,
+  bulkCreate,
+  services,
+  buildReasonMessage,
+  ruleExecutionLogger,
+  tuple,
+  alertSuppression,
+  wrapSuppressedSequences,
+  alertGroupFromSequenceBuilder,
+  addSequenceSuppressionTermsAndFields,
+  alertWithSuppression,
+  alertTimestampOverride,
+  experimentalFeatures,
+  mergeSourceAndFields = false,
+  maxNumberOfAlertsMultiplier,
+}: BulkCreateSuppressedSequencesParams) => {
+  const suppressOnMissingFields =
+    (alertSuppression?.missingFieldsStrategy ?? DEFAULT_SUPPRESSION_MISSING_FIELDS_STRATEGY) ===
+    AlertSuppressionMissingFieldsStrategyEnum.suppress;
+
+  const suppressibleWrappedSequences: Array<
+    WrappedFieldsLatest<EqlShellFieldsLatest & SuppressionFieldsLatest> & {
+      subAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+    }
+  > = [];
+  const unsuppressibleWrappedDocs: Array<WrappedFieldsLatest<BaseFieldsLatest>> = [];
+
+  sequences.forEach((sequence) => {
+    const alertGroupFromSequence = alertGroupFromSequenceBuilder(sequence, buildReasonMessage);
+    const shellAlert = alertGroupFromSequence?.[0];
+    if (shellAlert != null) {
+      if (!suppressOnMissingFields) {
+        const [shellAlertWithFields] = partitionMissingFieldsEvents(
+          shellAlert != null ? [shellAlert] : [],
+          alertSuppression?.groupBy || [],
+          ['fields'],
+          mergeSourceAndFields
+        );
+
+        if (shellAlertWithFields.length === 0) {
+          // unsuppressible sequence alert
+          // directly push alertGroup because these docs are wrapped already
+          unsuppressibleWrappedDocs.push(
+            ...(alertGroupFromSequence as Array<
+              | WrappedFieldsLatest<EqlShellFieldsLatest>
+              | WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>
+            >)
+          );
+        } else {
+          const wrappedWithSuppressionTerms = addSequenceSuppressionTermsAndFields(
+            shellAlert,
+            alertGroupFromSequence.slice(1) as Array<
+              WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>
+            >,
+            buildReasonMessage
+          );
+          suppressibleWrappedSequences.push(wrappedWithSuppressionTerms);
+        }
+      } else {
+        const wrappedWithSuppressionTerms = addSequenceSuppressionTermsAndFields(
+          shellAlert,
+          alertGroupFromSequence.slice(1) as Array<
```

Comment:
and here

Reply:
Fixed here: https://github.com/elastic/kibana/pull/189725/commits/3079b9eabcd2274281f1678704f1b09c5cc0e655